### PR TITLE
Loki shard splitting: enable for supported metric queries

### PR DESCRIPTION
**Pull request by matyax**
Queries wrapped with `sum()`, or only containing `sum_over_time()`, and `count_over_time()`, and `bytes_over_time()`, will be executed with shard splitting if the feature flag is enabled.

Additionally, we're removing the app limitation from the condition.

Otherwise, time splitting will be used.

### CODE REVIEW

File: public/app/plugins/datasource/loki/datasource.ts [hunk 1]
```diff
@@ -361,11 +361,7 @@ export class LokiDatasource
       return this.runLiveQueryThroughBackend(fixedRequest);
     }
 
-    if (
-      config.featureToggles.lokiShardSplitting &&
-      requestSupportsSharding(fixedRequest.targets) &&
-      fixedRequest.app === CoreApp.Explore
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/loki/datasource.ts:367]: does this mean sharding will fire in dashboard panels now?

**Review by gtk-grafana — APPROVED** [review]
LGTM

File: public/app/plugins/datasource/loki/datasource.ts (hunk 1 above)
Comment (matyax) [reply public/app/plugins/datasource/loki/datasource.ts:367]: Yes, that was the request, but only if the feature flag is enabled.

Comment (gtk-grafana) [reply public/app/plugins/datasource/loki/datasource.ts:367]: Sounds good, wasn't sure as the PR description doesn't mention that

Comment (matyax) [reply public/app/plugins/datasource/loki/datasource.ts:367]: Totally. Good call.

File: public/app/plugins/datasource/loki/queryUtils.ts [hunk 2]
```diff
@@ -319,11 +322,56 @@ export function requestSupportsSharding(allQueries: LokiQuery[]) {
...
+      return supportedWrappingVectorOpps.length > 0;
+    }
+  }
+
+  return true;
+}
+
+function getNodeString(query: string, node: SyntaxNode) {
+  return query.substring(node.from, node.to);
+}
```
Comment (svennergr) [inline public/app/plugins/datasource/loki/queryUtils.ts:373]: FWIW, there are a few helper methods in the `NodePosition` class that might make this a bit more readable: https://github.com/grafana/grafana/blob/b4a78a5cea8277a6a6e8c84424c6d16f9dc6f07b/public/app/plugins/datasource/loki/modifyQuery.ts#L48-L54

File: public/app/plugins/datasource/loki/datasource.ts (hunk 1 above)
Comment (svennergr) [reply public/app/plugins/datasource/loki/datasource.ts:367]: Should be very similar to time splitting, but did we do a quick sanity check how it behaves with transformations?

File: public/app/plugins/datasource/loki/queryUtils.ts (hunk 2 above)
Comment (matyax) [reply public/app/plugins/datasource/loki/queryUtils.ts:373]: We could extract NodePosition from `modifyQuery` to make it more visible.

File: public/app/plugins/datasource/loki/datasource.ts (hunk 1 above)
Comment (matyax) [reply public/app/plugins/datasource/loki/datasource.ts:367]: Good question. Yeah, same as time splitting.

**Pull request merged**